Controllers/WordsController.cs

```diff
@@ -63,5 +137,31 @@ public async Task<IActionResult> PostWord(Word word)
 
             return Ok(word);
         }
+
+        public async Task<List<Word>> AllWords()
+        {
+            var allWords = await _context.Words.ToListAsync();
+            return allWords;
+        }
+
+        public async Task<Word> GetWordById(Guid id)
+        {
+            var word = await _context.Words.FindAsync(id);
+
+            if (word == null)
+            {
+                throw new NullReferenceException();
```

Comment:
NullReferenceException should not be used like that) If you want to pass any custom errors, just create your own type of exception